src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs

```diff
@@ -410,55 +410,61 @@ private static ModuleInfo[] GetModuleInfos(int processId, bool firstModuleOnly)
-                        ret = Interop.mincore.GetModuleBaseName(processHandle, moduleHandle, baseName, baseName.Capacity * 2);
-                        if (ret == 0) throw new Win32Exception();
-                        moduleInfo._baseName = baseName.ToString();
+                        {
+                            HandleError();
+                            break;
+                        }
+
+                        ModuleInfo moduleInfo = new ModuleInfo
+                        {
+                            _sizeOfImage = ntModuleInfo.SizeOfImage,
+                            _entryPoint = ntModuleInfo.EntryPoint,
+                            _baseOfDll = ntModuleInfo.BaseOfDll
+                        };
 
-                        StringBuilder fileName = new StringBuilder(1024);
+                        int ret = Interop.mincore.GetModuleBaseName(processHandle, moduleHandle, baseName, baseName.Capacity * 2);
+                        if (ret == 0)
+                        {
+                            HandleError();
+                            break;
+                        }
+
+                        moduleInfo._baseName = baseName.ToString();
                         ret = Interop.mincore.GetModuleFileNameEx(processHandle, moduleHandle, fileName, fileName.Capacity * 2);
-                        if (ret == 0) throw new Win32Exception();
+                        if (ret == 0)
+                        {
+                            HandleError();
+                            break;
+                        }
+
                         moduleInfo._fileName = fileName.ToString();
 
                         if (moduleInfo._fileName != null
                             && moduleInfo._fileName.Length >= 4
                             && moduleInfo._fileName.StartsWith(@"\\?\", StringComparison.Ordinal))
                         {
-                            moduleInfo._fileName = moduleInfo._fileName.Substring(4);
+                            moduleInfo._fileName = fileName.ToString(4, fileName.Length - 4);
                         }
 
                         moduleInfos.Add(moduleInfo);
-                    }
-                    catch (Win32Exception e)
-                    {
-                        if (e.NativeErrorCode == Interop.mincore.Errors.ERROR_INVALID_HANDLE || e.NativeErrorCode == Interop.mincore.Errors.ERROR_PARTIAL_COPY)
-                        {
-                            // It's possible that another thread casued this module to become
-                            // unloaded (e.g FreeLibrary was called on the module).  Ignore it and
-                            // move on.
-                        }
-                        else
-                        {
-                            throw;
-                        }
+                        break;
                     }
 
+                    baseName.Clear();
+                    fileName.Clear();
```

Comment:
These should be at the top of the for loop so that we clear even if we skip some portion of the loop due to an error, right?